### PR TITLE
Update docs and errors to use XDG config path

### PR DESCRIPTION
**Pull request by Ralith**
As the more conventional path, users should be directed this way preferentially.

### CODE REVIEW

File: docs/config.md [hunk 1]
```diff
@@ -8,7 +8,7 @@ These are the config settings available to jj/Jujutsu.
 The config settings are loaded from the following locations. Less common ways to
 specify `jj` config settings are discussed in a later section.
 
-* `~/.jjconfig.toml` (global)
+* `~/.config/jj/config.toml` (global)
```
Comment (ilyagr) [inline docs/config.md:11]: You'd also want to reword the first paragraph of the " Alternative ways to specify configuration settings" section, around line 454.

Comment (ilyagr) [inline docs/config.md:11]: I'm worried the `~/.config/jj/config.toml` path might not actually be correct on Windows. I thing it uses some sort of a `%USER_DATA_SOMETHING_SOMETHING%` variable.

Comment (martinvonz) [reply docs/config.md:11]: Yes, that's why I changed (back) to `~/.jjconfig.toml` in commit 96849da332c809041040371b55eb5b4aaf097330. Now that we have this document, maybe we can describe the different locations here and then just try to say "in your config file" or something elsewhere. I'm not sure if that's better.

Comment (ilyagr) [reply docs/config.md:11]: It might be nice to make this clearer in config.md and the comments of config.rs.

Comment (Ralith) [reply docs/config.md]: Done.

Comment (Ralith) [reply docs/config.md]: Reworded things to make config.md the single source of truth.

File: cli/src/cli_util.rs [hunk 2]
```diff
@@ -1394,7 +1394,7 @@ See https://github.com/martinvonz/jj/blob/main/docs/working-copy.md#stale-workin
         {
             writeln!(
                 ui.warning(),
-                r#"Name and email not configured. Until configured, your commits will be created with the empty identity, and can't be pushed to remotes. To configure, add something like the following to $HOME/.jjconfig.toml:
+                r#"Name and email not configured. Until configured, your commits will be created with the empty identity, and can't be pushed to remotes. To configure, add something like the following to your config file:
   user.name = "Some One"
   user.email = "someone@example.com""#
```
Comment (martinvonz) [inline cli/src/cli_util.rs:1399]: Should we instead use `jj config set --user` here too?

```
Name and email not configured. Until configured, your commits will be created with the empty identity, and can't be pushed to remotes. To configure, run something like the following:
jj config set --user user.name = "Some One"
jj config set --user user.email = "someone@example.com"
```

Comment (Ralith) [reply cli/src/cli_util.rs:1399]: Ooh, I didn't know we had that! That's the best of both worlds.

File: docs/config.md [hunk 3]
```diff
@@ -8,11 +8,12 @@ These are the config settings available to jj/Jujutsu.
 The config settings are loaded from the following locations. Less common ways to
 specify `jj` config settings are discussed in a later section.
 
-* `~/.jjconfig.toml` (global)
+* [The global config file]
```
[the remark this answers is not in the document]
Comment (Ralith) [reply docs/config.md]: Went with "The user config file".

File: cli/src/cli_util.rs (hunk 2 above)
Comment (Ralith) [reply cli/src/cli_util.rs]: Done.

File: docs/config.md [hunk 4]
```diff
@@ -502,10 +503,13 @@ executable on your system](https://facebook.github.io/watchman/docs/install)).
 
 Debugging commands are available under `jj debug watchman`.
 
-# Alternative ways to specify configuration settings
+# Configuration file locations
 
-Instead of `~/.jjconfig.toml`, the config settings can be located under
-a platform-specific directory. It is an error for both of these files to exist.
+On all platforms, the user's global `jj` configuration file may located at
```
[the remark this answers is not in the document]
Comment (Ralith) [reply docs/config.md]: Went with "is located".

File: README.md [hunk 5]
```diff
@@ -243,11 +243,11 @@ cargo install --git https://github.com/martinvonz/jj.git --locked --bin jj jj-cl
...
-Create a file at `~/.jjconfig.toml` and make it look something like
+Create a file at `~/.config/jj/config.toml` and make it look something like
 this:
 
 ```shell script
-$ cat ~/.jjconfig.toml
+$ cat ~/.config/jj/config.toml
 [user]
 name = "Martin von Zweigbergk"
 email = "martinvonz@google.com"
```
[the remark this answers is not in the document]
Comment (Ralith) [reply README.md]: Done.

**Review by martinvonz — APPROVED** [review]

**Pull request merged**